qudipy/qutils/matrices.py

```diff
@@ -1,315 +1,809 @@
-"""
+'''
 Constant matrices used to define quantum gates
-
-@author: hromecB
-"""
+@author: hromecB, zmerino
+'''
+import os
+import inspect
 import numpy as np
+import json
+
+class Operator:
+    '''
+
+    Initialize Operator class. Handles all things realted to
+    creating/loading/modifying operator object files which contain various
+    operators in dictionary format. The Operator class also contains hard
+    coded operators.
+
+    '''
+
+    ##### Operator Object Manipulation #####
+
+    def __init__(self, operators=None, filename=None, f_type=None):
+        '''
+
+        Keyword Arguments
+        -----------------
+        lib: Dictionary
+            Dictionary of operators to be used to initialize an object.
+            If left empty then the dictionary is going to be built from scratch
+            or loaded from an operator object file from the working directory.
+        filename: string
+            String that indicates the name of the existing object file which
+            contains a library of operators.
+
+        Returns
+        -------
+        None.
+
+        '''
+
+        # Initialize object attribute for later use
+        self.is_unitary = None
+
+        # Ensure initial operator dictionary only contains items with
+        # the correct data structure, then assign to object
+        if filename is None:
+            if operators is None:
+                # Empty dictionary
+                self.lib = {}
+            elif operators is not None:
+                # Use user defined dictionary without existing object file
+                self.check_ops(operators)
+                self.lib = operators
+        elif  filename is not None:
+            
+            # Append '.npz' if only the filename with out file type suffix
+            if filename[-4:] != '.npz':
+                filename = f'{filename}.npz'
+
+            # Set file name of object instance so multiple library objects
+            # can be referenced
+            self.filename = filename
```

Comment:
This line means that the `self.filename` attribute is assigned only if there's an initial filename when creating an operator object. This manifests itself in issues with `__repr__()` function and some hardcoded operators that reference `save_ops(filename)` (these are the ones I spotted). 

To fix this, just move the code line to the left so that a filename (perhaps, 'None') is always saved
```suggestion
        self.filename = filename
```

Reply:
Thanks for catching this. This was a typo.